### DISCUSSION TITLE
Modifying `targetRevision` automatically

### DISCUSSION THREAD

**Discussion by djfinnoy** [argoproj/argo-cd#9710] [Q&A]
Consider a simple mono-repo for application manifests:

```
├── apps
│   ├── team-a
│   │   ├── app-1a
│   │   ├── app-2a
│   │   └── app-3a
│   └── team-b
│       ├── app-1b
│       ├── app-2b
│       └── app-3b
│
└── app-of-apps
    ├── dev
    │   ├── team-a.yaml
    │   └── team-b.yaml
    └── prod
        ├── team-a.yaml
        └── team-b.yaml
```

`apps/` contains the configuration for bespoke apps, their source code is kept in separate repositories.
`app-of-apps/` contains the `Application` manifests for the apps in `apps/`.

Now let's assume that a developer makes a pull request against `apps/team-a/app1-a` while working on the source code for app1-a.
It would be nice if the `targetRevision` of app1-a in `dev` automatically jumped to the pull request branch, and returned to it's default (eg. `main`) after the pull request is closed.

I can think of a couple of ways to do this, but they all have drawbacks / don't feel very elegant:
- [Tag tracking](https://argo-cd.readthedocs.io/en/stable/user-guide/tracking_strategies/#tag-tracking)
- Git writeback to `app-of-apps/dev/.argocd-source-<appname>.yaml` via CI
- ArgoCD `Application` patching via CI

How would you do it?

**Comment by kostis-codefresh** [comment]
The Pull Request generator exists precisely for this scenario https://argo-cd.readthedocs.io/en/latest/operator-manual/applicationset/Generators-Pull-Request/ Have you seen it?

Full example here https://github.com/kostis-codefresh/preview-env-manifests/blob/main/argocd/smart-pr-generator.yml
